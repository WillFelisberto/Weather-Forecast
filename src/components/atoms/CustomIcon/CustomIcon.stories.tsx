import type { Meta, StoryObj } from '@storybook/react';
import Image from 'next/image';

import svg from '../../../../public/icons/cloudy-2-day.svg';
import { CustomIcon } from './CustomIcon';

const meta: Meta<typeof CustomIcon> = {
  title: 'Atoms/CustomIcon',
  component: CustomIcon,
  tags: ['autodocs'],
  args: {
    iconSVG: <Image src={svg} alt="Ícone de nuvem parcialmente nublado" />,
    size: 64
  }
};

export default meta;

type Story = StoryObj<typeof CustomIcon>;

export const Default: Story = {};
