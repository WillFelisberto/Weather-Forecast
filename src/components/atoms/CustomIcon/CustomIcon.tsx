import React from 'react';

interface CustomIconProps {
  iconSVG: React.ReactElement; // O SVG completo para o tema claro
  size?: number; // Tamanho do ícone (em pixels)
}

export const CustomIcon: React.FC<CustomIconProps> = ({ iconSVG, size = 64 }) => {
  return (
    <div className="relative">
      {/* Ícone para tema claro */}
      {React.cloneElement(iconSVG as React.ReactElement<React.SVGProps<SVGSVGElement>>, {
        width: size,
        height: size
      })}
    </div>
  );
};
