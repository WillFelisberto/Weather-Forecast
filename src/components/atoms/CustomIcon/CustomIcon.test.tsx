import { render, screen } from '@/tests/test-utils';
import React from 'react';

import '@testing-library/jest-dom';
import { CustomIcon } from './CustomIcon';

describe('CustomIcon Component', () => {
  const sampleSVG = (
    <svg data-testid="custom-icon" viewBox="0 0 100 100" xmlns="http://www.w3.org/2000/svg">
      <circle cx="50" cy="50" r="40" fill="blue" />
    </svg>
  );

  it('renders the SVG with default size', () => {
    const { container } = render(<CustomIcon iconSVG={sampleSVG} />);

    const svgElement = screen.getByTestId('custom-icon');

    // Verifica se o SVG foi renderizado
    expect(svgElement).toBeInTheDocument();

    // Verifica se o tamanho padrÃ£o foi aplicado
    expect(svgElement).toHaveAttribute('width', '64');
    expect(svgElement).toHaveAttribute('height', '64');

    expect(container).toMatchSnapshot();
  });

  it('renders the SVG with a custom size', () => {
    render(<CustomIcon iconSVG={sampleSVG} size={128} />);

    const svgElement = screen.getByTestId('custom-icon');

    // Verifica se o tamanho customizado foi aplicado
    expect(svgElement).toHaveAttribute('width', '128');
    expect(svgElement).toHaveAttribute('height', '128');
  });
});
