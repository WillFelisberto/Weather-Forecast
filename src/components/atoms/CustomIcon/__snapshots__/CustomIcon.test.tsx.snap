// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomIcon Component renders the SVG with default size 1`] = `
<div>
  <div
    class="relative"
  >
    <svg
      data-testid="custom-icon"
      height="64"
      viewBox="0 0 100 100"
      width="64"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        cx="50"
        cy="50"
        fill="blue"
        r="40"
      />
    </svg>
  </div>
</div>
`;
